
When route is invoked with no arguments, fallback to "netstat -r"
to list existing routes.

--- a/src/sbin/route/route.c
+++ b/src/sbin/route/route.c
@@ -139,9 +139,19 @@
 usage(const char *cp)
 {
 	if (cp != NULL)
+	{
 		warnx("bad keyword: %s", cp);
 	errx(EX_USAGE, "usage: route [-dnqtv] command [[modifiers] args]");
 	/* NOTREACHED */
+	} else {
+		warnx("warning: route invoked with no arguments. fallback to \"netstat -r\"\n");
+
+		/* For compatibility with the Linux version of route. */
+		char *opt = nflag ? "-n" : NULL;
+		char *netstat_args[] = { "netstat", "-r", opt, NULL };
+		execvp("netstat", netstat_args);
+		err(1, "netstat");
+	}
 }
 
 int
